src/Neo/Network/P2P/Payloads/Signer.cs

```diff
@@ -66,6 +67,31 @@ public class Signer : IInteroperable, ISerializable
             /*AllowedGroups*/       (Scopes.HasFlag(WitnessScope.CustomGroups) ? AllowedGroups.GetVarSize() : 0) +
             /*Rules*/               (Scopes.HasFlag(WitnessScope.WitnessRules) ? Rules.GetVarSize() : 0);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool Equals(Signer other)
+        {
+            if (ReferenceEquals(this, other))
+                return true;
+            if (other is null) return false;
+            return Account == other.Account &&
+                Scopes == other.Scopes &&
+                AllowedContracts.SequenceEqual(other.AllowedContracts) &&
```

Comment:
Some of these values are null depending of the Scope

Reply:
> scope can not be null,

But `AllowedContracts`/`AllowedGroups`/`Rules` can be null sometimes, that's what Shargon says.

Reply:
Yes, some of the values can be null, we should make unit tests with real signers with different scopes, depending of the scope, you will find nulls entries

Reply:
`Xxxs.AsSpan().SequenceEqual(other.AsSpan())`.

`SequenceEqual` in` Sysmte.MemoryExtension`s is  null safe.

Reply:
![image](https://github.com/user-attachments/assets/4a2039f4-d37e-4fea-9197-28ffae89a9c7)